### ISSUE TITLE
Export sort method doesn't sort subdecks

### ISSUE THREAD

**Issue by wyohm** [Stvad/CrowdAnki#188]
I was trying to export sorted by guid to make for minimal diffs as notes are added by contributors and wasn't seeing any change from default. Tried various other sort methods (eventually including reverse order) and none had any effect. Finally tried exporting a deck without subdecks (my original efforts were using a deck where all the notes were in subdecks (by lessons in a course)) and the export sorting works as expected.

It *may* be sorting the subdecks themselves (order of the subdecks not the order of the notes within) by the "crowdanki_uuid" since I see them in sequence, but it's also the order that the subdecks are listed in Anki, so there's no change and I'm not sure where the order of the subdecks is coming from.

**Comment by aplaice** [comment]
Yes, you're right.  The contents of subdecks currently [aren't sorted at all](https://github.com/Stvad/CrowdAnki/blob/master/crowd_anki/export/anki_exporter.py#L39). :/  I'll look into this!

The subdecks themselves are ("iteratively"\*) sorted by [their names](https://github.com/Stvad/CrowdAnki/blob/master/crowd_anki/representation/deck_initializer.py#L32).

\* "iteratively"/recursively in the sense that if we have deck `a` with subdecks `b1` and `b2`, of which `b1` contains `c1` and `c2`, while `b2` contains `c3` and `c4`, then you'll have:

```json
{
  "name": "a"
  "children": [
    {
      "name": "b1",
      "children": [
        {
          "name": "c1",
          ...
        },
        {
          "name": "c2",
          ...
        },
      ],
      ...
    },
    {
      "name": "b2",
      "children": [
        {
          "name": "c3",
          ...
        },
        {
          "name": "c4",
          ...
        },
      ],
      ...
    }
  ],
  ...
}

```

Thanks!

**Comment by wyohm** [comment]
Thanks for explaining subdeck sorting!
